packages/bot-arbitrage/src/arbitarger.ts

```diff
@@ -0,0 +1,148 @@
+import { ArbParams, Market, MarketWithToken, Method } from "./types";
+import { ArbParamsStruct, MgvArbitrage } from "./types/typechain/MgvArbitrage";
+import { Mangrove, typechain, MgvToken } from "@mangrovedao/mangrove.js";
+
+export const activateTokens = async (
+  mgv: Mangrove,
+  tokens: MgvToken[],
+  arbitragerContract: MgvArbitrage
+) => {
+  const values = await Promise.all(
+    tokens.map((token) => {
+      return token.allowance({
+        owner: arbitragerContract.address,
+        spender: mgv.address,
+      });
+    })
+  );
+
+  const tokenToActivate = values
+    .map((value, index) => ({ value, index }))
+    .filter((value) => value.value.eq(0))
+    .map((value) => tokens[value.index]);
+
+  const gasEstimate = await arbitragerContract.estimateGas.activateTokens(
+    tokenToActivate.map((token) => token.address)
+  );
+
+  const tx = await arbitragerContract.activateTokens(
+    tokenToActivate.map((token) => token.address),
+    {
+      gasLimit: gasEstimate.mul(10),
+    }
+  );
```

Comment:
is it still needed in prod ? or is it jsut a workaround for the tests ?

Reply:
I needs for both I guess but I reduce it to multiply by two. It should be ok as we called this once per token